kufar/books/main.go

```diff
@@ -188,46 +192,45 @@ func (d Database) UpdateBook(w http.ResponseWriter, r *http.Request) {
 	id := mux.Vars(r)["id"]
 	idInt, err := strconv.Atoi(id)
 	if err != nil {
-		fmt.Printf("cant convert string: %v", err)
+		log.Printf("cant convert string: %v", err)
 	}
 
 	if id == "" {
-		http.Error(w, http.StatusText(400)+": missing parameter in url", http.StatusBadRequest)
+		http.Error(w, http.StatusText(http.StatusBadRequest)+": missing parameter in url", http.StatusBadRequest)
 		return
 	}
 
 	bk := Book{}
 
-	err = json.NewDecoder(r.Body).Decode(&bk)
-	if err != nil {
-		fmt.Printf("Error unmarshalling json: %v", err)
+	if err := json.NewDecoder(r.Body).Decode(&bk); err != nil {
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 	}
 
 	_, err = d.db.Exec("update books set id = $1, name = $2, author = $3, date = $4 where id = $1", idInt, bk.Name, bk.Author, bk.Date)
 	if err != nil {
-		http.Error(w, http.StatusText(500), http.StatusInternalServerError)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 		return
 	}
-	w.Write([]byte(http.StatusText(200)))
+	w.Write([]byte(http.StatusText(http.StatusOK)))
 }
 
 //DeleteBook ...
 func (d Database) DeleteBook(w http.ResponseWriter, r *http.Request) {
 	id := mux.Vars(r)["id"]
 	idInt, err := strconv.Atoi(id)
 	if err != nil {
-		fmt.Printf("cant convert string: %v", err)
+		log.Printf("cant convert string: %v", err)
```

Comment:
`http.Error` and `return`

Reply:
Sure, ill change it. Thought that since we're just converting a string. we didn't need to use `<http.Error>`

Reply:
@JustinObanor  Nope, it's an BadRequest error because your API expect `int` here

Reply:
@bynov Alright then